File: 03-frontend/angular-ecommerce/src/app/common/purchase.ts
import {Customer} from "./customer";
import {Address} from "./address";
import {Order} from "./order";
import {OrderItem} from "./order-item";

export class Purchase {

  customer: Customer;
  shippingAddress: Address;
  billingAddress: Address;
  order: Order;
  orderItems: OrderItem[];

}

File: spring-boot-ecomerce/src/main/java/com/freudromero/ecomerce/dto/Purchase.java
package com.freudromero.ecomerce.dto;

import com.freudromero.ecomerce.entity.Address;
import com.freudromero.ecomerce.entity.Customer;
import com.freudromero.ecomerce.entity.Order;
import com.freudromero.ecomerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}
